tree/dataframe/src/RSqliteDS.cxx

```diff
@@ -26,6 +32,328 @@
 #include <cstring>
 #include <stdexcept>
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/time.h>
+
+#include <linux/limits.h>
+
+namespace {
+
+constexpr char const* gSQliteVfsName = "Davix";
+
+struct VfsRootFile {
+   VfsRootFile(): pos(&c) {}
+   sqlite3_file pFile;
+   DAVIX_FD* fd;
+   uint64_t size;
+   Davix::Context c;
+   Davix::DavPosix pos;
+};
+
+static int VfsRdOnlyClose(sqlite3_file *pFile) {
+   Davix::DavixError *err = NULL;
+   VfsRootFile *p = reinterpret_cast<VfsRootFile*>(pFile);
+   if (p->pos.close(p->fd, &err) == -1){
+      p->~VfsRootFile();
+      return SQLITE_IOERR_CLOSE;
+   }
+   p->~VfsRootFile();
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyRead(
+   sqlite3_file *pFile,
+   void *zBuf,
+   int count,
+   sqlite_int64 offset
+) {
+   Davix::DavixError *err = NULL;
+   VfsRootFile *p = reinterpret_cast<VfsRootFile*>(pFile);
+   if (p->pos.pread(p->fd, zBuf, count, offset, &err) == -1) {
+      return SQLITE_IOERR;
+   }
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyWrite(
+   sqlite3_file * /*pFile*/,
+   const void * /*zBuf*/,
+   int /*iAmt*/,
+   sqlite_int64 /*iOfst*/)
+{
+   return SQLITE_OPEN_READONLY;
+}
+
+static int VfsRdOnlyTruncate(
+   sqlite3_file * /*pFile*/,
+   sqlite_int64 /*size*/)
+{
+   return SQLITE_OPEN_READONLY;
+}
+
+static int VfsRdOnlySync(
+   sqlite3_file * /*pFile*/,
+   int /*flags*/)
+{
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyFileSize(sqlite3_file *pFile, sqlite_int64 *pSize) {
+  VfsRootFile *p = reinterpret_cast<VfsRootFile*>(pFile);
+  *pSize = p->size;
+  return SQLITE_OK;
+}
+
+static int VfsRdOnlyLock(
+   sqlite3_file * /*pFile*/,
+   int /*level*/
+) {
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyUnlock(
+   sqlite3_file * /*pFile*/,
+   int /*level*/
+) {
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyCheckReservedLock(
+   sqlite3_file * /*pFile*/,
+   int *pResOut
+) {
+   *pResOut = 0;
+   return SQLITE_OK;
+}
+
+static int VfsRdOnlyFileControl(
+  sqlite3_file * /*p*/,
+  int /*op*/,
+  void * /*pArg*/
+) {
+  return SQLITE_NOTFOUND;
+}
+
+static int VfsRdOnlySectorSize(sqlite3_file *pFile __attribute__((unused))) {
+   return SQLITE_OPEN_READONLY;
+}
+
+static int VfsRdOnlyDeviceCharacteristics(sqlite3_file *pFile __attribute__((unused))) {
+   return SQLITE_OPEN_READONLY;
+}
+
+static int VfsRdOnlyOpen(
+   sqlite3_vfs * /*vfs*/,
+   const char *zName,
+   sqlite3_file *pFile,
+   int flags,
+   int * /*pOutFlags*/)
+{
```

Comment:
This and many functions above are not following our coding style for the function prototypes.
I also don't see the point in declaring these function `static` if they are already inside an unnamed namespace. Isn't that implied?

Reply:
I removed the `static` keyword from the function definitions. Regarding the prototype, as the functions are callback functions, the prototype is imposed by the sqlite library.

Reply:
As for the code style, this must have been picked up from older commits and is meanwhile resolved.